src/application.coffee

```diff
@@ -121,13 +121,20 @@ isValidPort = (port) ->
 	return parseFloat(port) is maybePort and maybePort > 0 and maybePort < 65535
 
 fetch = (app) ->
+	onProgress = (progress) ->
+		device.updateState(download_progress: progress.percent)
+
 	docker.getImage(app.imageId).inspectAsync()
 	.catch (error) ->
 		logSystemEvent(logTypes.downloadApp, app)
 		device.updateState(status: 'Downloading')
-		dockerUtils.fetchImageWithProgress app.imageId, (progress) ->
-			device.updateState(download_progress: progress.percentage)
-		.then ->
+
+		if app.env.RESIN_DELTA?
```

Comment:
It would be good to compare to a specific value and not only for the var's existence